### ISSUE TITLE
Copy Location Issue

### ISSUE THREAD

**Issue by eracknaphobia** [eracknaphobia/service.psvue.epg#29]
The epg default save location overwrites the epg.xml, playlist.m3u, and epg.db when the copy routines finish, resulting in blank files. Might need to set it up in away that the copy routine only runs when a certain mode is turned on, or the path is not the same as the plugin.video.psvue folder.

**Comment by eracknaphobia** [comment]
It might just be me, but I pulled in the new changes on my windows pc and that started happening. I wonder if @DiR3Wolf can test on his windows pc and see if it happens to him too.

**Comment by DiR3Wolf** [comment]
Same here, all epg, playlist and db are all blank. I ran them a few times to confirm.

**Comment by snoopyjoe** [comment]
Ok, so I made the file copy conditional if the default save to folder and the psvue.video folder are different. It's working on my end though I had to restart the service a couple times because it wasn't creating the data. One condition I did not test for is if someone changes the save location to the psvue.video folder. I don't think anyone would ever do this but in some cases the location paths would be different depending on the device and folder layout.

**Comment by DiR3Wolf** [comment]
@eracknaphobia @snoopyjoe It seems since the latest commit everything is working well, good enough for a new release?

I tested in on Windows, created a shared folder to copy the EPG files. Loaded those files on iptv simple on my laptop and had the full working channels/guide. I didn't try to play it because I didn't install the service.EPG on my laptop's Kodi, but assuming "slave" mode keeps the epg service from running, then it shouldn't be a problem.

One thing I noticed, EPG.db is also copied, in addition to EPG.xml and m3u file. It doesn't seem necessary. Maybe not a big deal if it doesn't tax the system much, but maybe something to consider later on when optimizing the code.

**Comment by snoopyjoe** [comment]
I wasn't sure if the epg.db would be accessed through simple pvr so I copied it too. But I think you're right and it's probably unnecessary. I'll test it this weekend and probably remove it.

Slave mode should only run the psvuewebservice then sys.exit. By passing all the guide data creation and refresh. So no data is created but the stream from the playlist can still be played. Glad it's working on your end.

**Comment by DiR3Wolf** [comment]
Yes, that's how understood slave mode worked, but I didn't get to test it. So assuming it behaves as intended, everything looks great.

**Comment by DiR3Wolf** [comment]
@eracknaphobia I believe this issue has been resolved. Should we close this issue?

**Comment by eracknaphobia** [comment]
Sorry, I've fallen behind on Kodi related stuff. I haven't tried testing the code yet, I will try to test this tonight when I get home.
